### PR TITLE
Added first register and login tests

### PR DESCRIPTION
**Pull request by amuresan9522**
-updated pom,xml file
-added HomePage, SignInPage, MyAccountPage, CreateAccountPage

Change-Id: I1cd8b85b432eb4919ea01435052fc09010a8b7bd

### CODE REVIEW

File: src/main/java/pages/CreateAccountPage.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class CreateAccountPage {
+    private WebDriver driver;
```
Comment (AlexK07) [inline src/main/java/pages/CreateAccountPage.java:7]: add a space between class name and driver

File: src/main/java/pages/CreateAccountPage.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    private By firstNameField = By.id("firstname");
+    private By lastNameField = By.id("lastname");
+    private By emailField = By.id("email_address");
+    private By passwordField = By.id("password");
+    private By passwordConfirmationField = By.id("password-confirmation");
+    private By createAccountAndSubmitButton = By.cssSelector(".action.submit.primary");
+
```
Comment (AlexK07) [inline src/main/java/pages/CreateAccountPage.java:14]: private final

File: src/main/java/pages/CreateAccountPage.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        driver.findElement(emailField).sendKeys(email);
+    }
+
+    public void inputPassword(String password) {
+        driver.findElement(passwordField).sendKeys(password);
+    }
+
+    public void inputPasswordConfirmation(String passconfirmation) {
+        driver.findElement(passwordConfirmationField).sendKeys(passconfirmation);
+    }
```
Comment (AlexK07) [inline src/main/java/pages/CreateAccountPage.java:37]: I would add a verb behind method names e.g.:  fill, type, send etc

File: src/main/java/pages/HomePage.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public class HomePage {
+    private WebDriver driver;
```
Comment (AlexK07) [inline src/main/java/pages/HomePage.java:8]: same line

File: src/main/java/pages/HomePage.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    private By greetingMessage = By.cssSelector("body > div.page-wrapper > header > div.panel.wrapper > div > ul > li.greet.welcome > span");
+    private By createAccountButton = By.xpath("/html/body/div[2]/header/div[1]/div/ul/li[3]/a");
+    private By signInButton = By.cssSelector("body > div.page-wrapper > header > div.panel.wrapper > div > ul > li.authorization-link > a");
```
Comment (AlexK07) [inline src/main/java/pages/HomePage.java:11]: final and we don't have other selectors for these? like id, class etc

File: src/test/java/BaseTest.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import pages.HomePage;
+
+import java.sql.SQLOutput;
```
Comment (AlexK07) [inline src/test/java/BaseTest.java:7]: this driver is for sql, do you need it ?

File: src/test/java/CreateAccountAndLoginTests.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        myAccountPage.clickSignOutFromMyAccountDropdown();
+    }
+
+    @Test
+    public void testSuccessfulLogin() {
+        SignInPage signInPage = homePage.clickOnSignInButton();
+        signInPage.inputSignInEmail("annaolsons@gmail.com");
+        signInPage.inputSignInPassword("HelloWorld22!");
+        HomePage homePage = signInPage.clickOnSignInButton();
+        //TODO: I need to wait a few seconds for the "default message to change to the correct message
```
Comment (AlexK07) [inline src/test/java/CreateAccountAndLoginTests.java:31]: you can create a wait helper class for these kind of actions

File: src/test/java/CreateAccountAndLoginTests.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+import pages.CreateAccountPage;
+import pages.HomePage;
+import pages.MyAccountPage;
+import pages.SignInPage;
+
+public class CreateAccountAndLoginTests extends BaseTest {
+    @Test
+    public void testSuccessfulRegister() {
+
+        CreateAccountPage createAccountPage = homePage.clickOnCreateAccountButton();
```
Comment (AlexK07) [inline src/test/java/CreateAccountAndLoginTests.java:14]: CreateAccountPage, MyAccountPage, SignInPage, HomePage can be moved to baseTest in beforeClass. In this way you don't need to declare them every time

**Review by AlexK07 — APPROVED** [review]
well done darling :*

File: src/main/java/pages/CreateAccountPage.java (hunk 3 above)
Comment (amuresan9522) [reply src/main/java/pages/CreateAccountPage.java:37]: I don't understand - do you mean to have it like firstNameInput? in my case, "input" is already the verb - has the same meaning as "fill"

File: src/main/java/pages/HomePage.java (hunk 4 above)
Comment (amuresan9522) [reply src/main/java/pages/HomePage.java:8]: What do you mean, to have private WebDriver driver in line 7? I have multiple private fields for the HomePage

File: src/main/java/pages/HomePage.java (hunk 5 above)
Comment (amuresan9522) [reply src/main/java/pages/HomePage.java:11]: For the create account button, yes, I will change it, I will also look for signInButton again. For the greetingMessage, not really, it;'s not a control but rather a text displayed.

File: src/test/java/BaseTest.java (hunk 6 above)
Comment (amuresan9522) [reply src/test/java/BaseTest.java:7]: oups, this looks like a copy-pasta code forgotten, will delete

File: src/test/java/CreateAccountAndLoginTests.java (hunk 7 above)
Comment (amuresan9522) [reply src/test/java/CreateAccountAndLoginTests.java:31]: Will try

File: src/test/java/CreateAccountAndLoginTests.java (hunk 8 above)
Comment (amuresan9522) [reply src/test/java/CreateAccountAndLoginTests.java:14]: ok

File: src/main/java/pages/HomePage.java (hunk 4 above)
Comment (amuresan9522) [reply src/main/java/pages/HomePage.java:8]: LE: Need to add a new blank line between line 7 and line 8

**Pull request merged**